crates/router/src/core/fraud_check.rs

```diff
@@ -588,51 +630,75 @@ pub async fn call_frm_before_connector_call<'a, F, Req, Ctx>(
-        {
-            if let Some(frm_configs) = frm_configs.clone() {
-                let mut updated_frm_info = make_frm_data_and_fraud_check_operation(
-                    db,
+    let (is_frm_enabled, frm_routing_algorithm, frm_connector_label, frm_configs) =
+        should_call_frm(merchant_account, payment_data, db, key_store.clone()).await?;
+    if let Some((frm_routing_algorithm_val, profile_id)) =
+        frm_routing_algorithm.zip(frm_connector_label)
+    {
+        if let Some(frm_configs) = frm_configs.clone() {
+            let mut updated_frm_info = make_frm_data_and_fraud_check_operation(
+                db,
+                state,
+                merchant_account,
+                payment_data.to_owned(),
+                frm_routing_algorithm_val,
+                profile_id,
+                frm_configs.clone(),
+                customer,
+            )
+            .await?;
+
+            if is_frm_enabled {
+                pre_payment_frm_core(
                     state,
                     merchant_account,
-                    payment_data.to_owned(),
-                    frm_routing_algorithm_val,
-                    profile_id,
-                    frm_configs.clone(),
+                    payment_data,
+                    &mut updated_frm_info,
+                    frm_configs,
                     customer,
+                    should_continue_transaction,
+                    should_continue_capture,
+                    key_store,
+                    operation,
                 )
                 .await?;
-
-                if is_frm_enabled {
-                    pre_payment_frm_core(
-                        state,
-                        merchant_account,
-                        payment_data,
-                        &mut updated_frm_info,
-                        frm_configs,
-                        customer,
-                        should_continue_transaction,
-                        should_continue_capture,
-                        key_store,
-                    )
-                    .await?;
-                }
-                *frm_info = Some(updated_frm_info);
             }
+            *frm_info = Some(updated_frm_info);
         }
-        logger::debug!("frm_configs: {:?} {:?}", frm_configs, is_frm_enabled);
-        return Ok(frm_configs);
     }
-    Ok(None)
+    let fraud_capture_method = frm_info.as_ref().and_then(|frm_info| {
+        frm_info
+            .frm_data
+            .as_ref()
+            .map(|frm_data| frm_data.fraud_check.payment_capture_method)
+    });
+    if matches!(fraud_capture_method, Some(Some(CaptureMethod::Manual)))
+        && matches!(
+            payment_data.payment_attempt.status,
+            api_models::enums::AttemptStatus::FrmUnresolved
+        )
+    {
+        payment_data.payment_intent.status = IntentStatus::RequiresCapture; // In Approve flow, payment which has payment_capture_method "manual" and attempt status as "FrmUnresolved",
```

Comment:
Payment data update should not happen within frm flow, only pass the suggestion on what to do with the transaction let the payments_core decide whether to consider the frm_suggestion or not. Check the update_trackers of payment_confirm

Reply:
Thanks for the info, I have added new frmSuggestion and moved this logic to approve->update trackers